c/main.c

```diff
@@ -0,0 +1,16 @@
+#include <stdlib.h>
+#include <unistd.h>
+
+#define lengthof(sstr) (sizeof (sstr) / sizeof *(sstr))
+#define sstr_len(sstr) (lengthof(sstr) - 1)
+#define sstr_unpack(sstr) (sstr), (sstr_len(sstr))
+
+static const char GREETING[] = "hello, world!\n";
+
+int main(void)
+{
+    return (
+        write(STDOUT_FILENO, sstr_unpack(GREETING))
+        == sstr_len(GREETING)
+    ) ? EXIT_SUCCESS : EXIT_FAILURE;
+}
```

Comment:
```suggestion
#include <stdio.h>

int main(int argc, char *argv[]) {
    printf("Hello from Aux!\n");
    return 0;
}
```
we really don't need to overcomplicate this

Reply:
`write` is lighter than `printf`. I think it is better to use it, and i am not sure by what means it is overcomplicated?
The macro are here to avoid the use of hard-coded value while providing compile-time information of the string size (which printf does not do).

Reply:
I agree with getchoo.
I think defining macros when it's arguably not needed is overcomplicating it. Also, I think we really don't care about the performance of printf vs write here. Getchoo's suggestion is the canonical hello world C program, it doesn't bring in any header other than stdlib, it doesn't define any item besides main (which you probably need), and it's recognizable so people know they should scrap it anyway.

I feel like the template should focus on nix code, and completely disregard C code (unless it demonstrates something on the nix side). I'd even be fine with an empty source file if that compiled.

Reply:
> write is lighter than printf

>> i feel this is once again putting a lot of focus on what is really just throwaway example code. i would really take a look at https://github.com/NixOS/templates/tree/master/c-hello as a reference

> and i am not sure by what means it is overcomplicated?

i don't think we're really going to get anywhere with this question being asked, so i would rather not continue with this

Reply:
I changed the implemtation to use a regular printf